src/imitation/algorithms/bc.py

```diff
@@ -62,7 +61,7 @@ def __init__(
         *,
         policy_class: Type[policies.BasePolicy] = base.FeedForward32Policy,
         policy_kwargs: Optional[Mapping[str, Any]] = None,
-        expert_dataloader: Optional[BCDataLoaderDucktype] = None,
+        expert_dataloader: Optional[types.DataLoaderInterface] = None,
```

Comment:
Why do we insist on a dataloader here but accept `types.Transitions` for `Adversarial`?

Where possible would be nice to move towards greater harmonization of their interfaces to simplify benchmarking different algorithms on the same environment.

Reply:
I'd support allowing `types.Transititions` as a convenience type here. I'm pretty sure I had implemented this choice at some point; don't remember why I removed it.

Reply:
Oh, that's because in `BC` we are missing a batch size variable which is needed to complete the default `DataLoader` constructor.

I think I'll add in a  "default batch size" class constant for this.

Reply:
Support for Transitions argument type added